### ISSUE TITLE
Consider using a Logging API

### ISSUE THREAD

**Issue by lucasstarsz** [lucasstarsz/FastJ-Game-Library#4]
`System.out` was good and all for the time, but this project is finally going somewhere -- might be time to give it some debugging wiiiiiiiiiiiiiiiiiiiiings.

**Comment by YeffyCodeGit** [comment]
Well, https://logging.apache.org/log4j/2.x/ is quite a popular logging API for Java, consider using that maybe?
And also I totally agree, and plus, logging information is a very useful tool in debugging.

**Comment by lucasstarsz** [comment]
The current idea I had was to make use of [**slf4j**](http://www.slf4j.org/). Since it allows people to add their own preferred logging backend while also providing the base infrastructure for logging that FastJ can use, it would likely be the best choice for overall compatibility and usability for everyone.

If you've any thoughts on the idea, feel free to comment.

**Comment by YeffyCodeGit** [comment]
Oh. I'm not that familiar with sl4j myself, but I have heard good things about it, so there both fair options.

**Comment by YeffyCodeGit** [comment]
If you would like, I could get started trying to write a simple implementation for logging?
I will need to familiarize myself with SL4J, but that shouldnt be that hard.

**Comment by lucasstarsz** [comment]
> If you would like, I could get started trying to write a simple implementation for logging?
> I will need to familiarize myself with SL4J, but that shouldnt be that hard.

Sure! Work at whatever pace you need. As long as we can get it done in a reasonable way, it'll work out pretty well. I'll update the issue with the plans that I would like to have completed before adding this to the game engine when I have some time 👍🏾

**Comment by YeffyCodeGit** [comment]
> Sure! Work at whatever pace you need. As long as we can get it done in a reasonable way, it'll work out pretty well. I'll update the issue with the plans that I would like to have completed before adding this to the game engine when I have some time 👍🏾

Of course, of course! I'll try to write something usable and PR. Always happy to help :D

**Comment by YeffyCodeGit** [comment]
I'm getting a bit burnt out with coding in general, so I will be un-assigning myself from this issue for now.

**Comment by lucasstarsz** [comment]
That's totally fine -- I was planning to handle it soon enough. Hope you feel better soon!
